libsolidity/lsp/Transport.h

```diff
@@ -45,6 +47,28 @@ enum class ErrorCode
 	RequestFailed = -32803
 };
 
+// Error exception used to bail out on errors in the LSP function-call handlers.
+class HandlerError: public util::Exception
+{
+public:
+	HandlerError(MessageID _id, ErrorCode _code, std::string _message):
+		m_id{_id},
+		m_code{_code},
+		m_message{_message}
+	{
+	}
+
+	MessageID id() const noexcept { return m_id; }
+	ErrorCode code() const noexcept { return m_code; }
+	std::string const& message() const noexcept { return m_message; }
+    char const* what() const noexcept override { return m_message.c_str(); }
```

Comment:
Do you actually need to implement this? Don't we have a clever base class?

Reply:
Also please do this in a separate pr

Reply:
Done in #12482.